datasets/subjqa/subjqa.py

```diff
@@ -0,0 +1,210 @@
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, f"SubjQA-master/SubjQA/{self.config.name}/splits/dev.csv")
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        df = pd.read_csv(filepath)
+        squad_format = self._convert_to_squad(df)
+        for example in squad_format["data"]:
+            title = example.get("title", "").strip()
+            for paragraph in example["paragraphs"]:
+                context = paragraph["context"].strip()
+                for qa in paragraph["qas"]:
+                    question = qa["question"].strip()
+                    question_meta = {k: v for k, v in qa.items() if k in self.question_meta_columns}
+                    id_ = qa["id"]
+                    answer_starts = [answer["answer_start"] for answer in qa["answers"]]
+                    answers = [answer["text"].strip() for answer in qa["answers"]]
+                    answer_meta = pd.DataFrame(qa["answers"], columns=self.answer_meta_columns).to_dict("list")
+                    yield id_, {
+                        **{
+                            "title": title,
+                            "context": context,
+                            "question": question,
+                            "id": id_,
+                            "answers": {
+                                **{
+                                    "answer_start": answer_starts,
+                                    "text": answers,
+                                },
+                                **answer_meta,
+                            },
+                        },
+                        **question_meta,
+                    }
+
+    def _create_paragraphs(self, df):
+        "A helper function to convert a pandas.DataFrame of (question, context, answer) rows to SQuAD paragraphs."
+        self.question_meta_columns = [
+            "domain",
+            "nn_mod",
+            "nn_asp",
+            "query_mod",
+            "query_asp",
+            "q_reviews_id",
+            "question_subj_level",
+            "ques_subj_score",
+            "is_ques_subjective",
+            "review_id",
+        ]
+        self.answer_meta_columns = ["answer_subj_level", "ans_subj_score", "is_ans_subjective"]
+        id2review = dict(zip(df["review_id"], df["review"]))
+        pars = []
+        for review_id, review in id2review.items():
+            qas = []
+            review_df = df.query(f"review_id == '{review_id}'")
+            id2question = dict(zip(review_df["q_review_id"], review_df["question"]))
+
+            for k, v in id2question.items():
+                d = df.query(f"q_review_id == '{k}'").to_dict(orient="list")
+                answer_starts = [eval(a)[0] for a in d["human_ans_indices"]]
```

Comment:
Please don't use `eval` since it's not safe. Instead you can use `ast.literal_eval`

Reply:
good catch! it's now fixed